Stanza_token/stanza_tokenization.py

```diff
@@ -0,0 +1,100 @@
+import stanza
+import langcodes
+import pandas as pd
+from datasets import load_dataset
+from tqdm import tqdm
+tqdm.pandas()
+pd.set_option("display.max_rows", 1000)
+pd.set_option("display.max_colwidth", None)
+
+dataset = load_dataset("CohereForAI/aya_dataset")
+df = pd.DataFrame(dataset["train"])
+df.drop_duplicates(inplace=True)
+
+
+# TODO: correct the file path
+openlid_pred = pd.read_csv(
+    "predictions/CohereForAI_aya_dataset_train_lid201_predictions.csv"
+)
+
+for column in ["inputs", "targets"]:
+    df[f"{column}_lid"] = openlid_pred[f"{column}_lid"].apply(lambda s: s[1:-1].split(":")[0])
+
+
+"""Define a function to get the language code from the language name,
+so as to allow Stanza to download the correct model"""
+def get_language_code(language_name):
+    try:
+        return langcodes.find(language_name).language
+    except langcodes.LanguageTagError:
+        return 'unknown'  # Handle languages that can't be found
+
+
+# Define functions to download and cache Stanza models
+def download_and_cache_models(language_codes):
+    pipelines = {}
+    for lang_code in language_codes:
+        try:
+            stanza.download(lang_code)
+            nlp_pipeline = stanza.Pipeline(lang=lang_code, processors='tokenize')
+            pipelines[lang_code] = nlp_pipeline
+        except Exception as e:
+            print(f"Stanza does not support language code '{lang_code}'. Error: {e}")
+    return pipelines
+
+# Define a function to tokenize text using Stanza
+# TODO: modify this to create a new column with the result, "stanza_used" or "not_stanza_used"
+def tokenize_with_stanza(text, lang_code, pipelines):
+    nlp_pipeline = pipelines.get(lang_code, None)
+    if nlp_pipeline:
+        doc = nlp_pipeline(text)
+        tokens = [token.text for sentence in doc.sentences for token in sentence.tokens]
+        return tokens
+    else:
+        # Fallback to a simple whitespace tokenizer if Stanza is not available
+        return text.split()
```

Comment:
Hmm, I am not sure if that's better than not attempting to tokenize text from these languages.

Reply:
Okay, I just thought that since Stanza wasn't 100% efficient for all languages and we were still trying to put together metrics on how to identify short prompts, we might want to leave the Python split for other languages until we find something better.

Reply:
On a second thought, I kept it as is, since I noticed you also flagged the samples that were tokenized using Stanza. So, one can have the choice to use whitespace tokenization for languages not supported by stanza, or completely ignore samples in these languages.